### PR TITLE
buildMozillaMach: use enableFeature instead of "inlining" it

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
To the record: the failure happened because [] has higher precedence than application function. Oh my.

**Comment by mweinelt** [comment]
The attribute for this file is `buildMozillaMach`.

**Comment by AndersonTorres** [comment]
@ofborg build firefox-unwrapped

**Comment by AndersonTorres** [comment]
@ofborg build buildMozillaMach

**Comment by mweinelt** [comment]
> The attribute for this file is `buildMozillaMach`.

Please use `buildMozillaMach:` instead of `firefox/common.nix`.

**Comment by AndersonTorres** [comment]
> Please use `buildMozillaMach:` instead of `firefox/common.nix:`.

You are talking about the commit message?

**Comment by AndersonTorres** [comment]
@mweinelt ping :)

**Comment by mweinelt** [comment]
While this reuses and existing function it certainly does not improve readability, so I'm not convinced.

**Comment by AndersonTorres** [comment]
It certainly does not worsen the readability, while it reuses an existing function.

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -370,25 +370,23 @@ buildStdenv.mkDerivation ({
...
+    (enableFeature geolocationSupport "necko-wifi")
+    (enableFeature gssSupport "negotiateauth")
+    (enableFeature webrtcSupport "webrtc")
+    (enableFeature crashreporterSupport "crashreporter")
+    (enableFeature debugBuild "debug")
+    (if debugBuild then "--enable-profiling" else "--enable-optimize" )
+    # --enable-release adds -ffunction-sections & LTO that require a big amount of
+    # RAM and the 32-bit memory space cannot handle that linking
+    (enableFeature (!debugBuild && !stdenv.is32bit) "release")
+    (enableFeature enableDebugSymbols "debug-symbols")]
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:389]: I think that makes it visually clearer where the list ends.

```suggestion
    (enableFeature enableDebugSymbols "debug-symbols")
  ]
```

**Comment by mweinelt** [comment]
> It certainly does not worsen the readability, while it reuses an existing function.

You say that now after reformatting it once more :wink:

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 2]
```diff
@@ -370,25 +370,23 @@ buildStdenv.mkDerivation ({
...
-  ++ flag crashreporterSupport "crashreporter"
-  ++ lib.optional (!drmSupport) "--disable-eme"
-
-  ++ (if debugBuild then [ "--enable-debug" "--enable-profiling" ]
-                    else [ "--disable-debug" "--enable-optimize" ])
-  # --enable-release adds -ffunction-sections & LTO that require a big amount of
-  # RAM and the 32-bit memory space cannot handle that linking
-  ++ flag (!debugBuild && !stdenv.is32bit) "release"
-  ++ flag enableDebugSymbols "debug-symbols"
+  ++ lib.optional (lib.versionAtLeast version "100") [(enableFeature sndioSupport "sndio")]
```
Comment (vcunat) [inline pkgs/applications/networking/browsers/firefox/common.nix:373]:
```suggestion
  ++ lib.optional (lib.versionAtLeast version "100") (enableFeature sndioSupport "sndio")
```

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 3]
```diff
@@ -370,25 +370,23 @@ buildStdenv.mkDerivation ({
...
-  ++ flag jackSupport "jack"
-  ++ flag pulseaudioSupport "pulseaudio"
-  ++ lib.optional (lib.versionAtLeast version "100") (flag sndioSupport "sndio")
-  ++ flag ffmpegSupport "ffmpeg"
-  ++ flag jemallocSupport "jemalloc"
-  ++ flag geolocationSupport "necko-wifi"
-  ++ flag gssSupport "negotiateauth"
-  ++ flag webrtcSupport "webrtc"
-  ++ flag crashreporterSupport "crashreporter"
-  ++ lib.optional (!drmSupport) "--disable-eme"
```
Comment (vcunat) [inline pkgs/applications/networking/browsers/firefox/common.nix:384]: This line gets dropped without replacement.  (I haven't checked for more yet.)

Comment (mweinelt) [reply pkgs/applications/networking/browsers/firefox/common.nix:384]: Seems to be the only missing line.

**Comment by AndersonTorres** [comment]
> You say that now after reformatting it once more 😉

I was still referring to the previous commit before force-pushing.

Now I am improving readability, in order to be more convincing.

File: pkgs/applications/networking/browsers/firefox/common.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/applications/networking/browsers/firefox/common.nix]: Done

File: pkgs/applications/networking/browsers/firefox/common.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/applications/networking/browsers/firefox/common.nix:384]: Done!

**Review by mweinelt — APPROVED** [review]

**Pull request merged**